internal/dag/builder_test.go

```diff
@@ -133,12 +133,31 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 		},
 	}
 
+	validClass := &gatewayapi_v1alpha1.GatewayClass{
```

Comment:
makes sense to only test against a valid+admitted GC at this level, will have to test weird combinations of Gateway+non-admitted GC at the controller level

Reply:
💯